entrypoint.sh

```diff
@@ -100,37 +99,139 @@ function super_lint() {
   /action/lib/linter.sh
 }
 
+function get_files_matching_filetype() {
+  local filetype=$1
+  shift
+
+  matching_files=()
+
+  for file in "$@"; do
+    filename=$(basename "$file")
+    
+    if [[ "$filename" == *"$filetype" ]]; then
+      matching_files+=("$file")
+    fi
+  done
+  echo "${matching_files[@]}"
+}
+
+function linter_failed() {
+  return=$1
+  shift 
+  name=$1
+
+  if [[ "${return:-1}" != 0 ]]; then
+    linter_exit_codes+=(["$name"]="${return}")
+  else
+    return "${return}"
```

Comment:
Hmm, what are we doing here? Maybe we move the `linter_failed` variable setting logic to here?

Reply:
But also, why do we not return anything explicitly if the exit code was non-zero?

Reply:
Refactored